hydra-gradle-core/src/main/java/com/pandora/hydra/LazyTestExcluder.java

```diff
@@ -24,19 +32,48 @@
     private final Supplier<HydraClient> hydraClient;
     private final String projectName;
     private final Project project;
+    private final Supplier<Set<String>> exclusionSupplier;
 
     private volatile Set<String> blacklist;
 
-    public LazyTestExcluder(Project project, Supplier<HydraClient> hydraClientSupplier) {
+    private LazyTestExcluder(Project project, Supplier<HydraClient> hydraClientSupplier, Supplier<Set<String>> exclusionSupplier) {
         this.hydraClient = hydraClientSupplier;
         this.projectName = project.getName();
         this.project = project;
+
+        if(exclusionSupplier == null) {
+            this.exclusionSupplier = this::fetchTestExcludesListFromHydraServer;
+        } else {
+            this.exclusionSupplier = exclusionSupplier;
+        }
+    }
+
+    public static LazyTestExcluder fromHydraServer(Project project, Supplier<HydraClient> hydraClientSupplier) {
+        return new LazyTestExcluder(project, hydraClientSupplier, null);
+    }
+
+    public static LazyTestExcluder fromExclusionFile(Project project, String pathToExclusionFile) {
+        Supplier<Set<String>> exclusionSupplier = () -> {
+            Path exclusionPath = Paths.get(pathToExclusionFile);
+            try {
+                return new LinkedHashSet<>(Files.readAllLines(exclusionPath));
+            } catch(IOException e) {
+                throw new GradleException("Unable to read exclusions from " + pathToExclusionFile);
+            }
+        };
+
+        return new LazyTestExcluder(project, () -> null, exclusionSupplier);
     }
 
     @Override
     public boolean isSatisfiedBy(FileTreeElement fileTreeElement) {
         if(blacklist == null) {
-            fetchTestExcludesListFromHydraServer();
+            synchronized(this) {
```

Comment:
Correct DCL with volatile field, but is it really needed here? I mean can you just init it in constructor and switch to final?

Reply:
We try and defer setting the blacklist until it is actually needed. If we set it in the constructor I think we would make a call to the hydra server at configuration time